src/sentry/search/snuba/backend.py

```diff
@@ -5,23 +5,35 @@
 import logging
 import math
 import pytz
-from collections import defaultdict
+from collections import defaultdict, OrderedDict
 from datetime import timedelta, datetime
 
 from django.utils import timezone
 
-from sentry.api.paginator import SequencePaginator
+from sentry.api.paginator import SequencePaginator, Paginator
 from sentry.event_manager import ALLOWED_FUTURE_DELTA
 from sentry.models import Release, Group, GroupEnvironment, GroupHash
 from sentry.search.django import backend as ds
 from sentry.utils import snuba
 from sentry.utils.dates import to_timestamp
+from sentry.utils.iterators import chunked
 
 
 logger = logging.getLogger('sentry.search.snuba')
 datetime_format = '%Y-%m-%dT%H:%M:%S+00:00'
 
 
+# maximum number of GroupHashes to send down to Snuba,
+# if more GroupHash candidates are found, a "bare" Snuba
+# search is performed and the result groups are then
+# post-filtered via queries to the Sentry DB
+MAX_PRE_SNUBA_CANDIDATES = 500
```

Comment:
I took 500 from the previous max we used to send to Snuba. (we would just silently trim searches)